### PR TITLE
Add reducers and actions

### PR DESCRIPTION
**Pull request by jellywiz**
This PR Configures and implements Redux toolkit

-  add redux
-  configure redux
-  add reducers for Books
-  add reducers for categories
-  implement reducers

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import { v4 as uuidv4 } from 'uuid';
+
+// Actions
+const ADD = 'Adding-Book';
+const REMOVE = 'Removing-Book';
```
Comment (ShadyShawkat) [inline src/redux/books/books.js]: - [ ] Kindly make sure that your action types are following the duck pattern, as per [this](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions.md#:~:text=Write%20your%20book%27s%20actions%20and%20reducer%20(in%20/src/redux/books/books.js)%20using%20the%20ducks%20pattern.) requirement.

- [HINT] The duck pattern should look something like the following:
```javascript
const <ACTION_TYPE_NAME> = '<APP_NAME>/<REDUCER_NAME>/<ACTION_TYPE>';
```

File: src/redux/categories/categories.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// actions
+const CHECK = 'Book-Check';
+
```
Comment (ShadyShawkat) [inline src/redux/categories/categories.js]: - [ ] Please do the same here.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @jellywiz, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by jellywiz** [comment]
Thank you @ShadyShawkat i will make sure it will be corrected

**Review by ShadyShawkat — APPROVED** [review]
## Hi @jellywiz, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by jellywiz** [comment]
Thanks @ShadyShawkat

**Pull request merged**